models/ldndc/R/write.config.LDNDC.R

```diff
@@ -163,99 +284,1837 @@ write.config.LDNDC <- function(defaults, trait.values, settings, run.id) {
+      else{
+        fractionalcover <- 0.5
+      }
+      # Fill in the value
+      eventsfile <- gsub("@FractionalCover@", paste0("'", fractionalcover, "'"), eventsfile)
+    }
+    
+    
+    ## Site file (general)
+    # Soil use history
+    if(any(grepl("@Info_Use_History@", sitefile))){
+      if ("history" %in% names(soil_IC_list$vals)){
+        soil_use_history <- unlist(soil_IC_list$vals["history"])[[1]]
+      }
+      else{
+        soil_use_history <- "arable"
+      }
+      sitefile <- gsub("@Info_Use_History@", paste0("'", soil_use_history, "'"), sitefile)
+    }
+    
+    # Soil type
+    if(any(grepl("@Soil_Type@", sitefile))){
+      if ("soil_type" %in% names(soil_IC_list$vals)){
+        soil_type <- unlist(soil_IC_list$vals["soil_type"])[[1]]
+      }
+      else{
+        soil_type <- "SALO"
+      }
+      sitefile <- gsub("@Soil_Type@", paste0("'", soil_type, "'"), sitefile)
+    }
+    
+    # Litter height
+    if(any(grepl("@Litter_Height@", sitefile))){
+      if ("litter_height" %in% names(soil_IC_list$vals)){
+        litter_height <- unlist(soil_IC_list$vals["litter_height"])[[1]]
+      }
+      else{
+        litter_height <- "0.0"
+      }
+      sitefile <- gsub("@Litter_Height@", paste0("'", litter_height, "'"), sitefile)
+    }
+    
+    ## -- Layers -- ##
+    
+    # Check how many depth layers is given and the depth of each
+    depth <- soil_IC_list$dims$depth
+    layer_count <- length(depth)
+    # Check what stratums is given for the layers
+    layer_div <- soil_IC_list$vals$stratum
+    
+    
+    for(depth_level in 1:layer_count){
+      
+      soil_one_block <- NULL
+      # Diskretization -- Every soil layer is still divided to several layers, this layer that contains these
+      # sublayers are here called a block. In LDNDC it is not suggested to use too tight layers so still will be
+      # divided to smaller layers that are not so thick.
+      
+      
+      
+      # For 1st level
+      if(depth_level == 1){ 
+        disk <- depth[depth_level] * 1000 / layer_div[depth_level]
+      }
+
+      # For rest of layers, depth is informed as cumulative, but LDNDC uses thickness
+      else{
+        disk <- (depth[depth_level] - depth[depth_level-1]) * 1000 / layer_div[depth_level]
+      }
+      
+      for(disk_level in 1:layer_div[depth_level]){
+        
+        
+        # Start creating a soil layer
+        soil_layer_values <- paste0("<layer depth='", disk, "' ")
+        
+        
+        ## Check which values are found from the soil netcdf file
+        if("soil_bulk_density" %in% names(soil_IC_list$vals)){
+          # Bulk density in a soil
+          bd <- PEcAn.utils::ud_convert(unlist(soil_IC_list$vals["soil_bulk_density"])[[depth_level]], "kg m-3", "kg dm-3")
+          soil_layer_values <- ifelse(!is.na(bd), paste0(soil_layer_values, paste0("bd='", bd, "' ")), soil_layer_values)
+        }
+        
+        if("soil_clay_content" %in% names(soil_IC_list$vals)){
```

Comment:
I'm curious where these soil variable names are coming from because they are not consistent with the PEcAn soil standard files, which are based on CF names (e.g. fraction_of_clay_in_soil). If your relying on model-specific files you probably need to add a conversion file that converts from PEcAn standard soil.nc to the model's file format and variable names. Here it looks like you're converting from something that's passed in to the function into the model-specific format, which seems to suggest that the thing being passed in should be PEcAn standard.

Reply:
Sounds good, @HenriKajasilta I think we should make these changes to the IC files we prepared.

But is there somewhere these are documented? Because I'm a bit confused, CF standard names use `mass_*` or `volume_fraction_of_clay_in_soil` for example, and the [documentation](https://pecanproject.github.io/pecan-documentation/develop/NewInput.html#soil-data-1) have an example like:

```
soil.data <- list(volume_fraction_of_sand_in_soil = c(0.3,0.4,0.5),
                  volume_fraction_of_clay_in_soil = c(0.3,0.3,0.3),
                  soil_depth = c(0.2,0.5,1.0))
                         
soil2netcdf(soil.data,"soil.nc")
```
but then [soil2netcdf](https://github.com/PecanProject/pecan/blob/develop/modules/data.land/R/soil2netcdf.R#L32) function only uses  `fraction_of_clay_in_soil` like you cite Mike. Further, `soil2netcdf` and `soil_params` functions use `soil_bulk_density` but CF name is just `soil_density`. I guess we should stick to the names in `soil2netcdf` and `soil_params`, right?

Reply:
I have changed the soil variable names in `write.config.LDNDC.R` to match the CF standard names. The variables for clay, sand and silt are now in from of `mass_* `and other changes concern soil density, field capacity, wilting point and hydraulic conductivity. I will make a new pull request as we also want the `soil2netcdf` and `soil_params` functions (and maybe some others) to follow the CF standards too.